### PR TITLE
Create snowflake Loader

### PR DESCRIPTION
**Pull request by sergiolrinditex**
Fixes # (issue)

#### Who can review?

Tag maintainers/contributors who might be interested:

### CODE REVIEW

**Comment by rlancemartin** [comment]
Thanks for the contribution! Minor nit on `lazy_load` vs `load` noted. 

Also, in the notebook how does a user specify their Snowflake DB to connect with? e.g., I didn't see credentials.

File: langchain/document_loaders/snowflakeLoader.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
+from __future__ import annotations
```
Comment (hwchase17) [inline langchain/document_loaders/snowflakeLoader.py:1]: naming of this file should be `snowflake_loader`

File: langchain/document_loaders/snowflakeLoader.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    def _get_columns(self, query_result: List[Dict[str, Any]]) -> Tuple[List[str], List[str]]:
+        page_content_columns = self.page_content_columns
+        metadata_columns = self.metadata_columns
+        if page_content_columns is None and query_result:
+            page_content_columns = query_result[0].keys()
+        if metadata_columns is None:
+            metadata_columns = []
+        return page_content_columns, metadata_columns
+
+    def load(self) -> List[Document]:
```
Comment (hwchase17) [inline langchain/document_loaders/snowflakeLoader.py:104]: per Lance's comment, lets implement lazy_load

**Review by hwchase17 — APPROVED** [review]

**Comment by rlancemartin** [comment]
@sergiolrinditex thanks for the contribution! I resolved all the line errors and merged. Please test whenever you get a chance (I don't have credentials needed to run notebook handy).

**Comment by rlancemartin** [comment]
@kaarthik108 contributed a fix:
https://github.com/hwchase17/langchain/pull/5967

**Pull request merged**